src/seedu/addressbook/commands/employee/EmployeeClockInCommand.java

```diff
@@ -54,17 +56,22 @@ private Attendance createNewAttendance(Attendance oldAttendance) {
 
     @Override
     public CommandResult execute() {
-        int index = rms.findAttendanceIndex(name);
+        try {
+            int index = rms.findAttendanceIndex(name);
 
-        Attendance oldAttendance = rms.findAttendance(index);
-        boolean isClockedIn = oldAttendance.getClockedIn();
-        if (isClockedIn) {
-            return new CommandResult(String.format(MESSAGE_NOT_YET_CLOCKED_OUT, name));
-        }
+            Attendance oldAttendance = rms.findAttendance(index);
+            boolean isClockedIn = oldAttendance.getClockedIn();
+            if (isClockedIn) {
+                return new CommandResult(String.format(MESSAGE_NOT_YET_CLOCKED_OUT, name));
+            }
 
-        Attendance newAttendance = createNewAttendance(oldAttendance);
+            Attendance newAttendance = createNewAttendance(oldAttendance);
 
-        rms.updateAttendance(oldAttendance, newAttendance);
-        return new CommandResult(String.format(MESSAGE_SUCCESS, name, this.currentDate));
+            rms.updateAttendance(oldAttendance, newAttendance);
+            return new CommandResult(String.format(MESSAGE_SUCCESS, name, this.currentDate));
+        }
+        catch (IndexOutOfBoundsException ie) {
+            return new CommandResult(Messages.MESSAGE_EMPLOYEE_NOT_IN_RMS);
```

Comment:
Error message should not be `MESSAGE_EMPLOYEE_NOT_IN_RMS`, it should be `MESSAGE_INVALID_EMPLOYEE_DISPLAYED_INDEX`